### PR TITLE
Refine Kotlin DSL release notes preamble & breaking changes section

### PR DESCRIPTION
**Pull request by eskatos**

### CODE REVIEW

File: subprojects/docs/src/docs/release/notes.md [hunk 1]
```diff
@@ -6,9 +6,11 @@ This will result in significantly reduced Java compilation time in subsequent bu
...
 If so, you'll be relieved to know that Gradle will now [periodically clean up unused `/caches`](#periodic-cache-cleanup) under `GRADLE_USER_HOME` and project root directories. 
 
-A moment you have anticipated is nearly here, as the [Kotlin DSL reaches version 1.0 RC1](https://github.com/gradle/kotlin-dsl/releases/tag/v1.0-RC1) with this release of Gradle.
-Configuration avoidance, script compilation caching, buildSrc refactoring propagation, and lots of DSL polish make this the release to try.
-_If you are interested in using the Kotlin DSL, please check out the [migration guide](https://guides.gradle.org/migrating-build-logic-from-groovy-to-kotlin/) and file issues in the [gradle/kotlin-dsl](https://github.com/gradle/kotlin-dsl) project._ 
+A moment you have anticipated is nearly here, as the [Kotlin DSL reaches version 1.0 RC1](https://github.com/gradle/kotlin-dsl/releases/tag/v1.0-RC1). 
+Configuration avoidance, `buildSrc` refactoring propagation to the IDE, and lots of DSL polish make this the release to try.
+The RC phase will span until the next Gradle version that will ship with the Gradle Kotlin DSL 1.0.
```
Comment (eriwen) [inline subprojects/docs/src/docs/release/notes.md]: Should we state this plainly "Gradle Kotlin DSL 1.0 will ship with the next version of Gradle"?

File: subprojects/docs/src/docs/release/notes.md [hunk 2]
```diff
@@ -138,9 +140,18 @@ These methods have been deprecated and the `create()` or `register()` methods sh
...
+The Kotlin DSL now enables the _SAM conversion for Kotlin functions_ Kotlin compiler feature in order to expose an uniform `org.gradle.api.Action<T>` based API to both Groovy DSL and Kotlin DSL scripts.
+
+The DSL types and behavior of containers elements delegated properties (e.g. `val jar by tasks`) and containers scopes (e.g. `tasks { }`) changed.
+
+The source set container can now be accessed using `project.sourceSets`, or just `sourceSets`.
+Previously it was located at `project.java.sourceSets`, or just `java.sourceSets`.  
+
+All these changes could cause script compilation errors.
+
+See the [release notes](https://github.com/gradle/kotlin-dsl/releases/tag/v1.0-RC1) for more information and how to fix builds broken by the changes described above.
```
Comment (eriwen) [inline subprojects/docs/src/docs/release/notes.md]: could mayyybe say "Kotlin DSL release notes" for the link. Kinda silly, but we see parts of release notes copied to blogs etc. and I weakly feel that we should fully qualify here.

File: subprojects/docs/src/docs/release/notes.md [hunk 3]
```diff
@@ -138,9 +140,18 @@ These methods have been deprecated and the `create()` or `register()` methods sh
 
 ## Potential breaking changes
 
-### Kotlin DSL breakages
+### Changes to the Gradle Kotlin DSL
 
-- `project.java.sourceSets` is now `project.sourceSets`
+The Kotlin DSL now enables the _SAM conversion for Kotlin functions_ Kotlin compiler feature in order to expose an uniform `org.gradle.api.Action<T>` based API to both Groovy DSL and Kotlin DSL scripts.
```
Comment (eriwen) [inline subprojects/docs/src/docs/release/notes.md]: This reads a bit too technical IMO. Can we simplify the message here a bit? I fear throwing around technical terms a lot indicates that this is complex (unjustly).

Comment (bamboo) [reply subprojects/docs/src/docs/release/notes.md]: We could leave some of the details out.

> This Kotlin DSL release takes advantage of experimental Kotlin compiler features in order to...

**Pull request merged**